query.go

```diff
@@ -1118,15 +1124,14 @@ func (s *ChainService) sendTransaction(tx *wire.MsgTx, options ...QueryOption) e
 	// it so we'll return the most rejected error between all of our peers.
 	//
 	// TODO(wilmer): This might be too naive, some rejections are more
-	// critical than others.
+	// critical than others. e.g. pushtx.Mempool and pushtx.Confirmed are OK.
 	//
 	// TODO(wilmer): This does not cover the case where a peer also rejected
 	// our transaction but didn't send the response within our given timeout
 	// and certain other cases. Due to this, we should probably decide on a
 	// threshold of rejections instead.
 	if numReplied == len(rejections) {
```

Comment:
this check actually also doesnt work cause `rejections` is a map... so this check only would work if each rejections error received was a unique one since this len will return the number of keys in the map

Reply:
Good point.  They'd all have to be unique errors, not so helpful.  Oh well.
Hopefully we see some rejected transactions logged to get an idea of when and if it's safe to throw and error back and remove the tx from the txn manager.